src/NHibernate/Util/ReflectHelperMemoryAssembly.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NHibernate.Util
+{
+	public class ReflectHelperMemoryAssembly
+    {
+		private static readonly ReflectHelperMemoryAssembly instance = new ReflectHelperMemoryAssembly();
+		private Dictionary<string, Assembly> _assemblyList;
+
+		public static ReflectHelperMemoryAssembly Instance
+		{
+			get
+			{
+				return instance;
+			}
+		}
+
+		/// <summary>
+		///  Remove all stored assembly in instance.
+		/// </summary>
+		public void Reset()
+		{
+			_assemblyList = new Dictionary<string, Assembly>();
```

Comment:
Once switched to a concurrent dictionary, `Clear` it instead, because assigning a new one is not thread safe. Of course clearing it all is still a bit debatable if other threads may add assemblies.